book/src/dev/rfcs/xxxx-block-subsidy.md

```diff
@@ -0,0 +1,207 @@
+
+## Founders reward
+
+Only functions specific to calculation of founders reward.
+
+- `founders_reward(Height, Network) -> Result<Amount<NonNegative>, Error>` - Founders reward portion for this block or 0.
+- `founders_reward_address(Height, Network) -> Result<String, Error>` - Address of the receiver founder at this block, address is a `String` because it can be a transparent or sapling address(After Heartwood/ZIP-213). 
+
+## Funding streams
+
+Only functions specific to the calculation of funding streams.
+
+- `funding_stream(height, newtork) -> Result<Amount<NonNegative>, Error>` - Funding stream portion for this block or 0.
+- `funding_stream_address(height, network) -> Result<String, Error>` - Address of the funding stream receiver at this block. The same as founders reward the returned address is a `String`.
+
+## Consensus rules
+
+All consensus rules for block subsidy are validated in a function inside `zebra-consensus/src/block/check.rs`:
+
+`subsidy_is_correct(Network, &Block) -> Result<(), BlockError>`
+
+The following consensus rules must be applied to `subsidy_is_correct()`:
+
+### 1 - Founders reward:
+
+*[Pre-Canopy] A coinbase transaction at `height` **MUST** include at least one output that pays exactly `FoundersReward(height)` zatoshi with a standard P2SH script of the form `OP_HASH160 FounderRedeemScriptHash (height) OP_EQUAL` as its scriptPubKey.* https://zips.z.cash/protocol/protocol.pdf#foundersreward
+
+We make use of the founders reward functions here. We get the amount of the reward with `founders_reward(height, network)` and we get the addresses for the reward at height with `founders_reward_address(height, network)`. Next we cab get a list of outputs that match the amount with `find_output_with_amount()` and with it we check if it matches the address.
+
+### 2 - Funding stream:
+
+*[Canopy onward]  The coinbase transaction at `height` **MUST**  contain at least one output per funding stream `fs` active at
+height, that pays `fs.Value(height)` zatoshi in the prescribed way to the stream’s recipient address represented by `fs.AddressList` of `fs.AddressIndex(height)`.* https://zips.z.cash/protocol/protocol.pdf#fundingstreams
+
+We make use of the funding streams functions here, similar to founders reward . We get the amount using `funding_stream(height, network)` and then the address with `funding_stream_address(height, network)`. The validation is again the same as with founders reward, we need to look for an output that pays the amount to the address. The same utility function `find_output_with_amount()` can be used here. 
+
+### 3 - Miner subsidy:
+
+*The transaction fees are the sum of each transactions inputs, minus its outputs, for all non-coinbase transactions in a block the sum of the coinbase transaction outputs must be less than or equal to the block subsidy plus transaction fees (because miners can destroy coins)* https://github.com/ZcashFoundation/zebra/pull/1051#issuecomment-700325653
```

Comment:
We need a better link here.

Reply:
> The total amount of transparent outputs from a coinbase transaction, minus the amount of the `valueBalance` field if present, MUST NOT be greater than the amount of miner subsidy, plus the total amount of transaction fees paid by transactions in this block
https://zips.z.cash/protocol/canopy.pdf#txnencodingandconsensus